pylearn2/datasets/tests/test_retina.py

```diff
@@ -0,0 +1,79 @@
+"""
+Tests for ../retina.py
+"""
+import numpy as np
+
+from pylearn2.utils import as_floatX
+from pylearn2.datasets import dense_design_matrix
+from pylearn2.datasets.retina import (foveate_channel,
+                                      defoveate_channel,
+                                      get_encoded_size,
+                                      encode,
+                                      decode,
+                                      RetinaEncodingBlock,
+                                      RetinaDecodingBlock,
+                                      RetinaCodingViewConverter
+                                      )
+
+
+class test_retina():
+    """
+    Parameters
+    ----------
+    None
+
+    Notes
+    -----
+    Testing class that checks retina preprocessing functions
+    """
+    def test_foveate_channel(self):
+        """
+        Foveate_channel implicitly tests downsample_ring and downsample_rect
+        """
+        img = np.array([[[0.51507119, 0.71686988], [0.62864236, 0.75365718]]])
+        output = np.zeros((1, 4))
+        foveate_channel(img, [2, 3], output, 0)
+        np.testing.assert_allclose(output, np.array([[0.65356015, 0.65356015,
+                                                      0., 0.]]))
+
+    def test_get_encoded_size(self):
+        """
+        Test encoding size method
+        """
+        assert get_encoded_size(4, 12, [2]) == 12
+
+    def test_encode(self):
+        """
+        Test encode method
+        """
+        test_input = np.array([[[[0, 0], [1, 1]]]])
+        np.testing.assert_array_equal(encode(test_input, [1]),
+                                      np.array([[0., 1., 0., 1.]]))
```

Comment:
The input has to be (batch, rows, cols, chans), so here it means that the image size is (1, 2), which is too small for a ring of 1, I think.
To make sure these make sense, you could use the same shape as for foveated norb: width=height=96, 2 channels, rings are [8, 4, 2, 2].